### PR TITLE
[CFM-1253] Create JS toggle for highlighting content

### PR DESCRIPTION
**Pull request by nirgn975**
https://issuetracker.amplexor.com/jira/browse/CFM-1253

### CODE REVIEW

**Comment by nirgn975** [comment]
To my future self:
If it's Sunday - Re-roll!

File: project/profiles/capacity4more/modules/c4m/content/c4m_content/c4m_content.module [hunk 1]
```diff
@@ -302,3 +302,75 @@ function c4m_content_current_context() {
     return 'global';
   }
 }
+
+/**
+ * Implements hook_field_extra_fields().
+ */
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/content/c4m_content/c4m_content.module]: Add elaborated docs

File: project/profiles/capacity4more/modules/c4m/content/c4m_content/c4m_content.module [hunk 2]
```diff
@@ -302,3 +302,75 @@ function c4m_content_current_context() {
...
+ *
+ * @param $node
+ *  The node to be changed.
+ * @param $status
+ *  The value to change to.
+ */
+function c4m_content_promote_change($node, $status) {
+  $wrapper = entity_metadata_wrapper('node', $node);
+
+  // Toggle the promote value.
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/content/c4m_content/c4m_content.module]: This is not a toggle

File: project/profiles/capacity4more/modules/c4m/content/c4m_content/c4m_content.module [hunk 3]
```diff
@@ -302,3 +302,75 @@ function c4m_content_current_context() {
...
+ */
+function c4m_content_menu() {
+  $items = [];
+
+  $items['foo/nojs/%node/%status'] = [
+    'title' => 'Change flag',
+    'page callback' => 'c4m_content_promote_change',
+    'page arguments' => [2, 3],
+    'access arguments' => array('access content'),
+    'access callback' => '',
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/content/c4m_content/c4m_content.module]: ?

File: project/profiles/capacity4more/modules/c4m/content/c4m_content/c4m_content.module [hunk 4]
```diff
@@ -302,3 +302,75 @@ function c4m_content_current_context() {
...
+ *  The value to change to.
+ */
+function c4m_content_promote_change($node, $status) {
+  $wrapper = entity_metadata_wrapper('node', $node);
+
+  // Toggle the promote value.
+  $wrapper->promote->set($status);
+  $wrapper->save();
+
+  return $node->title;
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/content/c4m_content/c4m_content.module]: remove

File: project/profiles/capacity4more/modules/c4m/content/c4m_content/c4m_content.module [hunk 5]
```diff
@@ -302,3 +302,75 @@ function c4m_content_current_context() {
...
+  );
+}
+
+/**
+ * Implements hook_menu().
+ */
+function c4m_content_menu() {
+  $items = [];
+
+  $items['foo/nojs/%node/%status'] = [
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/content/c4m_content/c4m_content.module]: `%status` => `%`

Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/content/c4m_content/c4m_content.module]: and of course change the path

File: project/profiles/capacity4more/modules/c4m/content/c4m_content/c4m_content.module [hunk 6]
```diff
@@ -395,3 +395,106 @@ function c4m_content_field_formatter_view($entity_type, $entity, $field, $instan
...
+  );
+}
+
+/**
+ * Implements hook_menu().
+ */
+function c4m_content_menu() {
+  $items = [];
+
+  $items['%/promote_node/%node/%/%'] = [
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/content/c4m_content/c4m_content.module]: $items['group/%node/promote_content/%node/%/%']

File: project/profiles/capacity4more/modules/c4m/content/c4m_content/c4m_content.module [hunk 7]
```diff
@@ -395,3 +395,106 @@ function c4m_content_field_formatter_view($entity_type, $entity, $field, $instan
...
+  }
+
+  $button = '<i class="fa fa-star fa-2x" aria-hidden="true"></i>';
+
+  drupal_add_library('system', 'drupal.ajax');
+  $query = array(
+      'tok' => drupal_get_token('delete_log_item' . $node->nid),
+    ) + drupal_get_destination();
+
+  $url = '/promote_node/'. $node->nid . '/' . !$node->promote;
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/content/c4m_content/c4m_content.module]:
```php
$url = '/group/' . $group->nid . '/promote_node/'. $node->nid . '/' . !$node->promote;
```

File: project/profiles/capacity4more/modules/c4m/content/c4m_content/c4m_content.module [hunk 8]
```diff
@@ -395,3 +395,106 @@ function c4m_content_field_formatter_view($entity_type, $entity, $field, $instan
...
+    return;
+  }
+
+  $button = '<i class="fa fa-star fa-2x" aria-hidden="true"></i>';
+
+  drupal_add_library('system', 'drupal.ajax');
+  $query = array(
+      'tok' => drupal_get_token('delete_log_item' . $node->nid),
+    ) + drupal_get_destination();
+
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/content/c4m_content/c4m_content.module]: `$group = og_context());`

**Comment by nirgn975** [comment]
`$output`:
capacity4more

File: project/profiles/capacity4more/modules/c4m/content/c4m_content/c4m_content.module [hunk 9]
```diff
@@ -395,3 +395,147 @@ function c4m_content_field_formatter_view($entity_type, $entity, $field, $instan
...
+/**
+ * Implements hook_field_extra_fields().
+ *
+ * Add a 'Highlighted button' to the 'Group Dashboard'.
+ */
+function c4m_content_field_extra_fields() {
+  $extra_fields['node']['group']['display']['highlighted'] = array(
+    'label' => t('Highlighted Button'),
+    'description' => t('Highlight a node will activate/deactivate his promoted field.'),
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/content/c4m_content/c4m_content.module]: => `t('Highlight a node will toggle its promote property.')`

File: project/profiles/capacity4more/modules/c4m/content/c4m_content/c4m_content.module [hunk 10]
```diff
@@ -395,3 +395,147 @@ function c4m_content_field_formatter_view($entity_type, $entity, $field, $instan
...
+    'weight' => 0,
+  );
+
+  return $extra_fields;
+}
+
+/**
+ * Implements hook_node_view().
+ *
+ * Adding the 'Highlighted' view for the promoted field.
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/content/c4m_content/c4m_content.module]: field => promote

File: project/profiles/capacity4more/modules/c4m/content/c4m_content/c4m_content.module [hunk 11]
```diff
@@ -395,3 +395,147 @@ function c4m_content_field_formatter_view($entity_type, $entity, $field, $instan
...
+ */
+function c4m_content_node_view($node, $view_mode, $langcode) {
+  if ($view_mode != 'group_dashboard') {
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/content/c4m_content/c4m_content.module]: remove this

File: project/profiles/capacity4more/modules/c4m/content/c4m_content/c4m_content.module [hunk 12]
```diff
@@ -395,3 +395,161 @@ function c4m_content_field_formatter_view($entity_type, $entity, $field, $instan
...
+      'class' => 'use-ajax',
+      'id' => 'highlight-content',
+    ),
+    'purl' => array('disabled' => TRUE),
+    'html' => TRUE,
+  );
+
+  // Determine if the the star type should be full or not by the highlight property.
+  $star_type = $highlight ? 'fa-star-o' : 'fa-star';
+  $start_tag = '<i class="fa ' . $star_type . ' fa-2x" aria-hidden="true"></i>';
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/content/c4m_content/c4m_content.module]: @nirgn975 
Should this one be `$star_tag` ?

**Comment by nirgn975** [comment]
`promote` field turn off:
capacity4more

`promote` field turn on:
capacity4more

**Comment by nirgn975** [comment]
@ordavidil Ready for review.

**Comment by nirgn975** [comment]
@ordavidil The test is fail because `Line exceeds 80 characters; contains 90 characters`, but it's in `c4m_user_og.module` file. The the commit before yours don't show this warning. And this file hasn't changed in this PR.
What should I do ?

**Pull request merged**